frontend_app_gallery/src/component/MainPreview.jsx

```diff
@@ -2,32 +2,88 @@ import React from 'react'
 import PropTypes from 'prop-types'
 import classnames from 'classnames'
 
-class MainPreview extends React.Component {
+export class MainPreview extends React.Component {
   constructor (props) {
     super(props)
-
+    this.imgContainer = React.createRef()
+    this.img = React.createRef()
     this.state = {
-      height: 0,
-      width: 0,
-      imageLoaded: false
+      imageLoaded: false,
+      style: {
+        maxWidth: '100%',
+        maxHeight: '100%',
+        width: '',
+        height: ''
+      }
     }
   }
 
   onLoad ({ target: img }) {
     this.setState({
-      height: img.height,
-      width: img.width,
       imageLoaded: true
     })
   }
 
-  render () {
-    const { props, state } = this
+  fixImageSize () {
```

Comment:
I would like a more precise name for this function

I'm not entirely sure about everything it does but I would call it something like fixImageRatioForRotation

Reply:
Yes this function name isn't precise enough,

I think `fixImageRatioForRotation` is a good name but I would prefer `compute` to `fix` because we don't really fix something that is broken.